java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanTask.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.dataset.scanner;
+
+import java.util.Iterator;
+
+import org.apache.arrow.vector.VectorSchemaRoot;
+
+/**
+ * Read record batches from a range of a single data fragment. A
+ * ScanTask is meant to be a unit of work to be dispatched. The implementation
+ * must be thread and concurrent safe.
+ */
+public interface ScanTask {
+
+  /**
+   * Creates and returns a {@link Itr} instance.
+   */
+  Itr scan();
+
+  /**
+   * The iterator implementation for {@link VectorSchemaRoot}s.
+   */
+  interface Itr extends Iterator<VectorSchemaRoot>, AutoCloseable {
```

Comment:
can we call this "Iterator"?

Reply:
Using Iterator is OK but it may collide with java.util.Iterator in scenarios. I we rename it should it be "BatchIterator" or something?

Reply:
renamed

Reply:
I would expect ScanTask.Iterator would be used consistently to avoid collision.  BatchIterator is also code with me.